clients/vscode-hlasmplugin/src/hlasmExternalFiles.ts

```diff
@@ -38,13 +40,20 @@ interface ExternalReadFileResponse {
     id: number,
     data: string,
 }
+
 interface ExternalReadDirectoryResponse {
     id: number,
     data: {
         members: string[],
         suggested_extension: string | undefined,
     },
 }
+
+type ExternalRequestDetails<R, L> = {
+    [ExternalRequestType.read_file]: R,
+    [ExternalRequestType.read_directory]: L,
+};
```

Comment:
why not use

```ts
interface ExternalRequestDetails<R, L> {
    [ExternalRequestType.read_file]: R;
    [ExternalRequestType.read_directory]: L;
};
```
?

Reply:
OK, I looked a bit more into this and it appears to me that the intention is like this:

```ts
// define a function operating on types that picks 2nd or third argument based on the first argument
type PurposeToArgs<
    Purpose extends ExternalRequestType,
    ReadArgs extends ClientUriDetails,
    ListArgs extends ClientUriDetails
> = Purpose extends ExternalRequestType.read_file ? ReadArgs : ListArgs;
//...
// then wherever currently there is 
ExternalRequestDetails<ReadArgs, ListArgs>[typeof purpose]
// use 
PurposeToArgs<typeof purpose, ReadArgs, ListArgs>
// instead
```

Reply:
I tried using just an inline expression, but for some reason ts did not like that...